yocto/myencfs_git.bb

```diff
@@ -0,0 +1,46 @@
+SUMMARY = "myencfs example application"
+DESCRIPTION="A simple directory encryption/decryption using authenticated encryption"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://COPYING;md5=9859f02baa869af387514f9e5dd8cc3e \
+                    file://debian/copyright;md5=358aa3ad2f63e0c87d59d3150ef3fd7e"
+
+SRC_URI = "git://github.com/alonbl/myencfs.git;protocol=https"
+
+SRCREV = "${AUTOREV}"
+S = "${WORKDIR}/git"
+
+# Crypto can be either openssl_crypto or mbedtls_crypto
+MYENCFS_CRYPTO ?= "openssl_crypto"
+# List of enabled features that can be overriden
+PACKAGECONFIG ?= "tools encrypt decrypt bio-file test fuse static-libs ${MYENCFS_CRYPTO}"
+
+
+PACKAGECONFIG[tools]	= "--enable-tool"
+PACKAGECONFIG[encrypt]	= "--enable-encrypt"
+PACKAGECONFIG[decrypt]	= "--enable-decrypt"
+PACKAGECONFIG[bio-file]	= "--enable-bio-file"
+PACKAGECONFIG[test]	= "--enable-tests"
+PACKAGECONFIG[static-libs]	= "--enable-all-static"
+PACKAGECONFIG[openssl_crypto]	= " \
+				--with-crypto=openssl, \
+				, \
+				openssl, \
+				openssl, \
+				, \
+				mbedtls_crypto \
+				"
+
+PACKAGECONFIG[mbedtls_crypto] = " \
+				--with-crypto=mbed, \
+				, \
+				mbedtls, \
+				mbedtls, \
+				, \
+				openssl_crypto \
+				"
+
+PACKAGECONFIG[fuse] = "--enable-fuse --enable-decrypt --enable-bio-file, \
```

Comment:
how do we produce an error if both openssl and mbed were specified?

Reply:
This is the magic of PACKAGECONFIG. The sisxth param is the "confilcts with" parameter. See for example crypto_openssl conflicts with crypto_mbedtls. If both would be enabled, the build will faill with a specific message.

Reply:
oh... how do you conflict with multiple? where can I read about it? I can find only this[1] source.

[1] https://www.yoctoproject.org/docs/2.4.2/ref-manual/ref-manual.html#var-PACKAGECONFIG

Reply:
According to documentation you can specify multiple conflicts as you can specify multiple depends or rdepends. I'll test that when I add wolfssl. As you probably saw PACKAGECONFIG functionality has been upgraded over the yocto releases, Initially, as you specified it has only supported 4 params.